src/qutip_cupy/dense_functions.py

```diff
@@ -29,3 +29,30 @@ def trace_cupydense(cp_arr):
     # @TODO: whnen qutip allows it we should remove this call to item()
     # as it takes a time penalty commmunicating data from GPU to CPU.
     return cp.trace(cp_arr._cp).item()
+
+
+def isherm_cupydense(cp_arr, tol):
+    if cp_arr.shape[0] != cp_arr.shape[1]:
+        return False
+    size = cp_arr.shape[0]
+    hermcheck_kernel = cp.RawKernel(
```

Comment:
This `cp.RawKernel` object should go outside of this function so that you don't pay the cost of recreating the same RawKernel object every time this function is called.